x/claims/keeper/ibc_callbacks_ibc_suite_test.go

```diff
@@ -132,13 +132,8 @@ func (suite *IBCTestingSuite) TestOnAcknowledgementPacketIBC() {
 			"correct execution - Claimed transfer",
 			func(claimableAmount int64) {
 				amt := sdk.NewInt(claimableAmount)
-				coins := sdk.NewCoins(sdk.NewCoin("aevmos", amt))
 
 				suite.chainA.App.(*app.Evmos).ClaimsKeeper.SetClaimsRecord(suite.chainA.GetContext(), senderAddr, types.ClaimsRecord{InitialClaimableAmount: amt, ActionsCompleted: []bool{true, true, true, true}})
-
-				// update the escrowed account balance to maintain the invariant
-				err := testutil.FundModuleAccount(suite.chainA.App.(*app.Evmos).BankKeeper, suite.chainA.GetContext(), types.ModuleName, coins)
```

Comment:
Why is it moved to [x/recovery/keeper/ibc_callbacks_integration_test.go](https://github.com/evmos/evmos/pull/786/files#diff-38e0dc67a93f5cee8058dadb16b89bdcc8865a6109e87f15b5ffc5a3b64549f8)

Reply:
yes, this one is very tricky. There is change in how ibc test works.

So, in this line, it calls SendAndReceiveMessage() which will invoke a series of IBC functions: https://github.com/evmos/evmos/blob/main/x/recovery/keeper/ibc_callbacks_integration_test.go#L108

Eventually, it will lead to this functions CommitBlock(): https://github.com/cosmos/ibc-go/blob/v3.2.0/testing/coordinator.go#L182 

And, also NextBlock(): https://github.com/cosmos/ibc-go/blob/v3.2.0/testing/chain.go#L266

For CommitBlock() and NextBlock(),
In IBC version 3.x, the flow would be: Commit -> BeginBlock.
In IBC version 5.x, the flow would be: EndBlock -> Commit -> BeginBlock.

EndBlock() is when CrisisKeeper will check all registered invariants. The current test logic of evmos will fail these invariants according to the new IBC test. 

For this line of code: https://github.com/evmos/evmos/blob/9e8f807883e6ecc4f9b8955103eceae9d5744713/x/recovery/keeper/ibc_callbacks_integration_test.go#L106. 
What I understand is that, the claim amount of 100aevmos is registered, all actions are false, so claims module should still have 100aevmos left but infact it is 0aevmos. In ibc v3, EndBlock() is not called first so no problem. However, in ibc v5, Endblock() is called first which triggers invariants check but because it has 0aevmos so it returns error.

The same goes for above in which all actions are true, so claims module is supposed to exclude 100aevmos. You can try it for yourself.